plugins/parsers/xpath/parser.go

```diff
@@ -108,7 +109,9 @@ func (p *Parser) Parse(buf []byte) ([]telegraf.Metric, error) {
 		}
 		if len(selectedNodes) < 1 || selectedNodes[0] == nil {
 			p.debugEmptyQuery("metric selection", doc, config.Selection)
-			return nil, fmt.Errorf("cannot parse with empty selection node")
+			if !p.AllowEmptySelection {
+				return metrics, fmt.Errorf("cannot parse with empty selection node")
+			}
```

Comment:
This also needs to be in the `ParseLine` function below (lines 152, 153)...

Reply:
As said before, the `ParseLine` only accepts 1 config (don't know why), so this setting wouldn't make much sense.

Reply:
😕 If the selector in parseLine doesn't return any nodes, it will not return a metric anyway. Enabling or disabling this new flag will not change the number of metrics emitted for ParseLine. The only difference would be wether or not an error would be added in the logfile. Is that what you want? (To possibly disable the error message in ParseLine?)

Reply:
We discussed on Slack to rewrite the ParseLine method to also support multiple configs. This has now been implemented.